### PR TITLE
docs: Update issue template

### PR DESCRIPTION
**Pull request by barbeau**
I've noticed that developers aren't providing some useful information in reports, like whether the issue they've seen occurs on a real device or emulator, and whether it can be reproduced using the demo app. I've updated the issue template to leave clear space for these details, and also removed some text that doesn't seem relevant to AMU.

I've also added clearer headings for expected vs. observed behavior.

@arriolac If any of this doesn't seem appropriate for this project, let me know and I can remove it.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -24,14 +24,28 @@ Check for answers on StackOverflow with the [google-maps](http://stackoverflow.c
...
+What did you see instead?  Describe your issue in detail here.
+
 #### Environment details
 
-1. Specify the API at the beginning of the title (for example, "Places: ...")
-2. OS type and version
-3. Library version and other environment information
+1. Device make and model, or emulator details
+1. Android version
+1. Library version and other environment information
```
Comment (arriolac) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Renumber as 1, 2, & 3?

Comment (barbeau) [reply .github/ISSUE_TEMPLATE/bug_report.md]: When this is rendered GitHub will convert these to actual numbers - it's the lazy persons way of numbering lists so you don't have to renumber then when items change 😉:

~~~
1. First
1. Second
1. Third
~~~

1. First
1. Second
1. Third

Comment (barbeau) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Although the developer will see these unrendered 🤦‍♂. I'll change it now...

Comment (barbeau) [reply .github/ISSUE_TEMPLATE/bug_report.md]: Renumbered in https://github.com/googlemaps/android-maps-utils/pull/667/commits/d97174181d4db7f8f7673c51da3ff06a010c95e9

**Pull request merged**